Added e2e tests for `post.scheduled` event

jennyxchang opened the pull request:
refs #15537
Added e2e test for post.scheduled event to increase webhook test coverage

jennyxchang committed:
Added e2e test for post.scheduled event

refs #15537
Added e2e test for post.scheduled event to increase webhook test coverage

naz commented on ghost/core/test/e2e-webhooks/posts.test.js:
```diff
@@ -154,4 +154,57 @@ describe('post.* events', function () {
+        scheduledPost.status = 'scheduled';
+        const publishedTime = new Date();
+        publishedTime.setDate(publishedTime.getDate() + 1);
+        scheduledPost.published_at = publishedTime.toISOString();
```
For date calculations in the codebase there's a preference to use a date library - [like here](https://github.com/TryGhost/Ghost/blob/34439f2e28e4603e9ef30c86954099d56a7c9ef2/ghost/core/test/regression/api/admin/schedules.test.js#L36).

The following: 
```js
moment().add(30, 'seconds').toDate()
```

is easier to read and later migrate to a different library (we have been trying to [migrate away to Luxon from momentjs](https://github.com/TryGhost/Ghost/issues/13648)).

jennyxchang replied:
Thanks for the suggestion! I updated the date calculation using the date library.

naz reviewed (changes requested):
Just a minor change request to use a library instead of manual date calculation. The rest looks 👌👌👌 Thanks @jennyxchang!

jennyxchang committed:
Fixed date calculation in e2e test for post.scheduled event

refs #15537
Used momentjs date library to calculate date, so that the code is easier to read and can later migrate to a different date library.

jennyxchang commented:
Thanks for reviewing! I updated the date calculation as requested.

naz reviewed (approved):
These changes are good and ready to be merged. Thank you @jennyxchang 🙌

ErisDS commented:
Hi @jennyxchang  thanks so much for this PR. Sorry for the delay, it has now been merged 🎉 and will appear in the next release of Ghost - usually Fridays.

I'm not sure if you found this through hacktoberfest, but I've added the accepted label to this PR to make sure it counts. We have plenty of open issues and ongoing projects for hacktoberfest and beyond. We would love to see you around again 👋
